### PR TITLE
Built in sast scans sto 7417

### PR DESCRIPTION
**Pull request by douglas-j-bothwell**
### SME review links (updated 2024-05-02)

- STO release notes > [UI update 1.15.0](https://6633cb3baae3301d82f538b7--harness-developer.netlify.app/release-notes/security-testing-orchestration/#ui-update-1150) - new built-in SAST scanner, support for Semgrep orchestration.
- Scanners supported by STO > [Code Repo scanners](https://6633cb3baae3301d82f538b7--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/security-step-settings-reference/#code-repo-scanners) - added orchestration support to Semgrep open-source and paid
- Built-in scanners > [SAST scan step reference](https://6633cb3baae3301d82f538b7--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/built-in/sast) 
- [Semgrep scanner reference](https://6633cb3baae3301d82f538b7--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference) - Added Semgrep licensing info to Important Notes, added workflow descriptions, miscellaneous tweaks/updates
-  [Tutorial: SAST code scans using Semgrep](https://6633cb3baae3301d82f538b7--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/sast-scan-semgrep) - Updated to use Orchestration workflow

### Peer review links

- STO release notes > [Version 1.93.0](https://662ae1f74d7ac92f6e1f1de0--harness-developer.netlify.app/release-notes/security-testing-orchestration/#version-1930) - new built-in SAST scanner, support for Semgrep orchestration
- Semgrep scanner reference > [Important notes for running Semgrep scans in STO](https://662ae1f74d7ac92f6e1f1de0--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference#important-notes-for-running-semgrep-scans-in-sto) - Added Semgrep licensing info
- Semgrep scanner reference > [Set-up workflows](https://662ae1f74d7ac92f6e1f1de0--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference#set-up-workflows)
- Semgrep scanner reference > formatting/terminology tweaks
  - [Target and variant detection](https://662ae1f74d7ac92f6e1f1de0--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference#target-and-variant-detection)
  - [Access Token](https://662ae1f74d7ac92f6e1f1de0--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference#access-token)
-  [Tutorial: SAST code scans using Semgrep](https://66324f7100a82fe046889a55--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/sast-scan-semgrep) - Updated to use Orchestration workflow

### Updates based on PC feedback

- [Scanners supported by STO](https://66324f7100a82fe046889a55--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/security-step-settings-reference/#scanners-supported-by-sto) - Added orchestration support for Semgrep and added Semgrep (paid version) to list of commercial SAST scanners
- [Semgrep settings reference](https://66324f7100a82fe046889a55--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference)
   - Revised intro to remove Semgrep description
   - Added link to updated SAST tutorial, commented out links to build-scan-push tutorial (will update this in a separate PR)
   - Updated [YAML pipeline example](https://66324f7100a82fe046889a55--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference#yaml-pipeline-example) to use Orchestration
- STO Release Notes > [Version 1.93.0](https://66324f7100a82fe046889a55--harness-developer.netlify.app/release-notes/security-testing-orchestration/#version-1930) - Added link to [SAST scan step reference](66324f7100a82fe046889a55--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/built-in/sast)
-  [Tutorial: SAST code scans using Semgrep](https://66324f7100a82fe046889a55--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/sast-scan-semgrep) - Updated to use Orchestration workflow

### Preview links

- STO release notes > [Version 1.93.0](https://662ae1f74d7ac92f6e1f1de0--harness-developer.netlify.app/release-notes/security-testing-orchestration/#version-1930) - new built-in SAST scanner, support for Semgrep orchestration
- STO release notes > [Important notes for running Semgrep scans in STO](https://662ae1f74d7ac92f6e1f1de0--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference#important-notes-for-running-semgrep-scans-in-sto) - Added Semgrep licensing info
- Semgrep scanner reference > [Set-up workflows](https://662ae1f74d7ac92f6e1f1de0--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference#set-up-workflows)
- Semgrep scanner reference > minor word tweaks
  - [Target and variant detection](https://662ae1f74d7ac92f6e1f1de0--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference#target-and-variant-detection)
  - [Access Token](https://662ae1f74d7ac92f6e1f1de0--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference#access-token)

### Prepub checks

PRs must meet these requirements to be merged:

- [x] Successful preview build.
- [x] SME review.
- [x] PM review.
- [x] Editorial review.
- [x] No merge conflicts.
- [x] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md [hunk 1]
```diff
@@ -13,28 +16,151 @@ The following tutorials include detailed examples of how to run a [Semgrep scan]
 
 ## Important notes for running Semgrep scans in STO
 
-<!--
+- This integration uses the [Semgrep Engine](https://github.com/semgrep/semgrep), which is is open-source and licensed under [LGPL 2.1](https://tldrlegal.com/license/gnu-lesser-general-public-license-v2.1-(lgpl-2.1)).
```
Comment (SmritiSatya) [inline docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md:19]: This integration uses the [Semgrep Engine](https://github.com/semgrep/semgrep), which is open-source and licensed under [LGPL 2.1](https://tldrlegal.com/license/gnu-lesser-general-public-license-v2.1-(lgpl-2.1)).

File: docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md [hunk 2]
```diff
@@ -13,28 +16,151 @@ The following tutorials include detailed examples of how to run a [Semgrep scan]
...
+:::
+
+To scan a code repository, you need [Harness Code Repository](/docs/code-repository) or a [Harness connector](/docs/category/code-repo-connectors) to your Git service. 
+
+
+#### Add the built-in SAST scanner
+
+Do the following:
+
+1. Add a Build or Security stage to your pipeline.
```
Comment (SmritiSatya) [inline docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md:63]: Add a **Build** or **Security** stage to your pipeline. (If this us relevant, you can use it in all places.)

File: docs/security-testing-orchestration/sto-techref-category/shared/step_palette/target/auto-detect/_app-instance.md [hunk 3]
```diff
@@ -1,10 +1,10 @@
-When auto-detect is enabled for application instances, the step detects these values as follows: 
+When **Detect** is enabled for application instances, the step detects these values as follows: 
   - The target is based on the **Instance Domain** and **Path** defined in the step or runtime input, for example `https://qa.jpgr.org:3002/login/us`.
   - The variant is the UTC timestamp when the step scanned the instance.
 
 Note the following:
-- Auto-detection is not available when the **Scan Mode** is Ingestion. 
-- Auto-detect is the default selection for new pipelines. Manual is the default for old pipelines, but you might find that neither radio button is selected in the UI.
+- **Detect** is not available when the **Scan Mode** is Ingestion.
```
Comment (SmritiSatya) [inline docs/security-testing-orchestration/sto-techref-category/shared/step_palette/target/auto-detect/_app-instance.md:6]: **Ingestion** (in bold, if it is a UI element.)

File: release-notes/security-testing-orchestration.md [hunk 4]
```diff
@@ -24,6 +24,23 @@ These release notes describe recent changes to Harness Security Testing Orchestr
...
+This release introduces a new built-in SAST step that adds a preconfigured [Semgrep step](/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference) that's ready to run as soon as you add it. (STO-6738) 
+
+This step is behind the feature flag `STO_ONE_CLICK_SAST`. Contact [Harness Support](mailto:support@harness.io) to enable it.
+
+<DocImage path={require('/docs/security-testing-orchestration/sto-techref-category/semgrep/static/add-built-in-sast-scanner.png')} width="50%" height="50%" title="Add built-in SAST scan step" /> 
+
+#### New feature: Orchestrated Semgrep scans
+
+Harness STO now supports running [Semgrep scans](/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference) in Orchestration mode. (STO-3525)
+
```
Comment (SmritiSatya) [inline release-notes/security-testing-orchestration.md:40]: Harness STO now supports running [Semgrep scans](/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference) in **Orchestration** mode. (STO-3525)

**Review by SmritiSatya — APPROVED** [review]

File: docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md [hunk 5]
```diff
@@ -5,36 +5,156 @@ sidebar_label: Semgrep settings reference
...
+For a quick introduction, go to the [SAST code scans using Semgrep](/docs/security-testing-orchestration/sto-techref-category/semgrep/sast-scan-semgrep) tutorial.
 
-<!--
+## Important notes for running Semgrep scans in STO
 
-### Docker-in-Docker requirements
+- This integration uses the [Semgrep Engine](https://github.com/semgrep/semgrep), which is open-source and licensed under [LGPL 2.1](https://tldrlegal.com/license/gnu-lesser-general-public-license-v2.1-(lgpl-2.1)). For information about the security rulesets supported by different Semgrep products, go to the [Semgrep Registry](https://semgrep.dev/explore). 
 
-import StoDinDRequirements from '/docs/security-testing-orchestration/sto-techref-category/shared/dind-bg-step.md';
+  To run scans using a licensed version of [Semgrep Code](https://semgrep.dev/products/semgrep-code), add your Semgrep token in the [Access token](#access-token) field.
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md:19]:
```suggestion
  To run scans using a licensed version of [Semgrep Code](https://semgrep.dev/products/semgrep-code), add your Semgrep token in the [Access token](#access-token) field. 
```

File: release-notes/security-testing-orchestration.md [hunk 6]
```diff
@@ -24,6 +24,23 @@ These release notes describe recent changes to Harness Security Testing Orchestr
 
 ## April 2024
 
+### Version 1.93.0
+
+#### Early Access feature: Built-in SAST scans
```
Comment (brian-f-harness) [inline release-notes/security-testing-orchestration.md:29]:
```suggestion
#### Early access feature: Built-in SAST scans
```

File: docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md [hunk 7]
```diff
@@ -5,36 +5,156 @@ sidebar_label: Semgrep settings reference
...
+  To run scans using a licensed version of [Semgrep Code](https://semgrep.dev/products/semgrep-code), add your Semgrep token in the [Access token](#access-token) field.   
 
-<StoDinDRequirements />
+- If you want to add trusted certificates to your scan images at runtime, you need to run the scan step with root access. 
 
--->
+  You can set up your STO scan images and pipelines to run scans as non-root and establish trust for your proxies using custom certificates. For more information, go to [Configure STO to Download Images from a Private Registry](/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry).
 
-### Root access requirements 
+- The following topics contain useful information for setting up scanner integrations in STO:
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md:25]:
```suggestion
```

File: docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md [hunk 8]
```diff
@@ -5,36 +5,156 @@ sidebar_label: Semgrep settings reference
...
 
-import StoRootRequirements from '/docs/security-testing-orchestration/sto-techref-category/shared/root-access-requirements-no-dind.md';
+  - [What's supported in STO](/docs/security-testing-orchestration/whats-supported)
+  - [Security Testing Orchestration FAQs](/docs/faqs/security-testing-orchestration)
+  - [Optimize STO pipelines](/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/optimize-sto-pipelines)
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md:29]:
```suggestion
```

File: docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md [hunk 9]
```diff
@@ -5,36 +5,156 @@ sidebar_label: Semgrep settings reference
...
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md:30]: These are dup links under For more information in the preview

https://662ae1f74d7ac92f6e1f1de0--harness-developer.netlify.app/docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference/#important-notes-for-running-semgrep-scans-in-sto

File: docs/security-testing-orchestration/sto-techref-category/semgrep/sast-scan-semgrep.md [hunk 10]
```diff
@@ -16,252 +16,66 @@ redirect_from:
...
-        file: /shared/scan_results/semgrep.sarif
-```
-
-</TabItem>
-</Tabs>
 
 ## Run the pipeline and check your results
 
-1. In the Pipeline Studio, select **Run** (top right).
+1. Run the pipeline: click **Run** (top right). The **Run Pipeline** dialog box appears.
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/sto-techref-category/semgrep/sast-scan-semgrep.md:70]:
```suggestion
1. Run the pipeline: click **Run** (top right). The **Run Pipeline** dialog box appears.
```

File: docs/security-testing-orchestration/sto-techref-category/semgrep/sast-scan-semgrep.md [hunk 11]
```diff
@@ -16,252 +16,66 @@ redirect_from:
...
+1. Run the pipeline: click **Run** (top right). The **Run Pipeline** dialog box appears. 
 
-2. When prompted, enter your runtime inputs.
+2. Enter the repository and branch to scan.
 
-   - Under **Codebase**, enter the repository and branch to scan.
+   When you scan a codebase for the first time, the standard practice is to scan the root branch. This is usually the `main` or `master` branch.
 
-   - Under **Stage: \<_stage_name_>**, enter the [target name and variant](/docs/security-testing-orchestration/get-started/key-concepts/targets-and-baselines) you want to use.
+3. Run the pipeline and then wait for the execution to finish.
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/sto-techref-category/semgrep/sast-scan-semgrep.md:76]:
```suggestion
3. Run the pipeline and then wait for the execution to finish.
```

File: docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md (hunk 5 above)
Comment (brian-f-harness) [inline docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md:19]:
```suggestion
  To run scans using a licensed version of [Semgrep Code](https://semgrep.dev/products/semgrep-code), add your Semgrep token in the [Access token](#access-token) field. 
```

File: docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md [hunk 12]
```diff
@@ -5,36 +5,156 @@ sidebar_label: Semgrep settings reference
...
+## Important notes for running Semgrep scans in STO
 
-### Docker-in-Docker requirements
+- This integration uses the [Semgrep Engine](https://github.com/semgrep/semgrep), which is open-source and licensed under [LGPL 2.1](https://tldrlegal.com/license/gnu-lesser-general-public-license-v2.1-(lgpl-2.1)). For information about the security rulesets supported by different Semgrep products, go to the [Semgrep Registry](https://semgrep.dev/explore). 
 
-import StoDinDRequirements from '/docs/security-testing-orchestration/sto-techref-category/shared/dind-bg-step.md';
+  To run scans using a licensed version of [Semgrep Code](https://semgrep.dev/products/semgrep-code), add your Semgrep token in the [Access token](#access-token) field.   
 
-<StoDinDRequirements />
+- If you want to add trusted certificates to your scan images at runtime, you need to run the scan step with root access.
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/sto-techref-category/semgrep/semgrep-scanner-reference.md:21]:
```suggestion
- If you want to add trusted certificates to your scan images at runtime, you need to run the scan step with root access.
```

**Review by brian-f-harness — APPROVED** [review]
LGTM

**Review by atmollo-harness — DISMISSED** [review]

**Pull request merged**